dosfsck: remove MAP_POPULATE flag for mmap

jaysim opened the pull request:
When mmap() is called with the MAP_POPULATE flag in certain kernel versions (specially tested on 5.4 and 6.11), the kernel does not return if the opened device is removed.
Therefore, remove the MAP_POPULATE flag and handle the SIGBUS signal that occurs when accessing the memory mapped address afterwards.

jaysim committed:
dosfsck: remove MAP_POPULATE flag for mmap

When mmap() is called with the MAP_POPULATE flag in certain kernel versions
(specially tested on 5.4 and 6.11), the kernel does not return
if the opened device is removed.
Therefore, remove the MAP_POPULATE flag and handle the SIGBUS signal
that occurs when accessing the memory mapped address afterwards.

Signed-off-by: JaeHoon Sim <jay.sim@lge.com>

hclee reviewed (commented):
Looks good to me.

jaysim committed:
dosfsck: fix checking media value of both FATs

During changing to a new function,
(commit: 5cbe32a44a1fe3d8dbf9b6800defeeb348354e5e)
checking media value of FATs was implemented differently from previous
code. Fix it.

Signed-off-by: JaeHoon Sim <jay.sim@lge.com>

jaysim commented:
> Looks good to me.

I've add one more patch to fix checking media value from both FATs. https://github.com/fatprogs/fatprogs/pull/8/commits/dde5dd60d46bc152fe2f9236d0f5ee9f7088df65
Could you review for it ?

hclee commented:
> > Looks good to me.
> 
> I've add one more patch to fix checking media value from both FATs. [dde5dd6](https://github.com/fatprogs/fatprogs/commit/dde5dd60d46bc152fe2f9236d0f5ee9f7088df65) Could you review for it ?

Looks good to me

jaysim committed:
fatprogs: Release 2.13.3

fatprogs v2.13.3 - released 2022-7-18
=====================================

BUG FIXES:
  * dosfsck: fix the incorrect media type comparison
  * dosfsck: remove the MAP_POPULATE flag from the mmap() call
  * dosfsck: fix the attempt to clean the volume dirty flag
              even when using the '-n' option.
  * dosfsck: fix volume lable check missing
  * test: add testing script for corruted images

Signed-off-by: JaeHoon Sim <jay.sim@lge.com>
